server/src/main/java/org/opensearch/cluster/coordination/CoordinationState.java

```diff
@@ -574,7 +580,7 @@ public void handlePrePublish(ClusterState clusterState) {
         // This is to ensure the remote store is the single source of truth for current state. Even if the current node
         // goes down after sending the cluster state to other nodes, we should be able to read the remote state and
         // recover the cluster.
-        if (isRemoteStateEnabled) {
```

Comment:
@soosinha lets not tie RemoteStateEnabled & publication. Basically a cluster can operate with just remote state enabled for durability and publication disabled.

Reply:
Leave this to RemoteStateEnabled. You can add a validation that for publication to be enabled, state has to be enabled.